### ISSUE TITLE
Highlander deployment failure using kubernetes v2 provider

### ISSUE THREAD

**Issue by danigar** [spinnaker/spinnaker#5077]
### Issue Summary:
**Highlander** deployments fail with the following error message: `Patch failed: The ReplicaSet '' is invalid. spec.template.metadata.labels: Invalid value: .......  selector does not match template labels`.

It happens after running a first **Highlander** deployment. The first one works fine, and the previous infraestructure is successfully destroyed.

### Cloud Provider(s):
Kubernetes running in GKE with kubernetes V2 provider.

### Environment:
Trying to deploy an application in a different kubernetes cluster where spinnaker is deployed. Using deployment manifest (helm) with `spinnaker traffic management` enabled.

### Feature Area:
Kubernetes v2 provider

### Description:
**Highlander** deployments fail with the following error message: `Patch failed: The ReplicaSet '' is invalid. spec.template.metadata.labels: Invalid value: .......  selector does not match template labels`.

First deployment using **Highlander** strategy works fine (same for **red/black** strategy). But after the first deployment, all deployments fail with the same message: `Patch failed: The ReplicaSet '' is invalid. spec.template.metadata.labels: Invalid value: .......  selector does not match template labels`.

I'm trying to deploy two different applications using `kubernetes v2 deployment (manifest)`. The applications have one service, one replicaset and istio stuff. I've tried to use arbitrary labels in the service selectors as suggested on the guidelines and here: https://github.com/spinnaker/spinnaker/issues/4923. But nothing of this works.

### Steps to Reproduce:
1. Deploy the application using a simple deploy (manifest) pipeline.
2. Deploy another version of the application using other deploy pipeline that manages traffic based on **Highlander** strategy.
3. Try to deploy other versions of the application using the same pipeline with the **Highlander** strategy. They always fail with the message quoted above.

### Additional Details:
Trying different strategies (**red/black**) we get the same behaviour.

<details>
  <summary>The deployment task success on the same pipeline</summary>

<img width="952" alt="Screenshot 2019-10-30 at 10 05 08" src="https://user-images.githubusercontent.com/6217436/67843804-fcb60280-fafc-11e9-920c-a8f70478dba5.png">
</details>

<details>
  <summary>The disable fails with the `Monitor disable` task.</summary>

<img width="955" alt="Screenshot 2019-10-30 at 10 04 57" src="https://user-images.githubusercontent.com/6217436/67843887-1e16ee80-fafd-11e9-9aeb-245764801714.png">
</details>

**Comment by danigar** [comment]
I think I've found the cause of the error (at least in my case). I will close the issue and left this comment here trying to help anyone that could be facing the same scenario. 
When you use Spinnaker managed traffic strategies to deploy any application, Spinnaker manages traffic adding the selector label of the service to the new deployed replica set. Spinnaker will do this through a patch operation using `json strategy`. My problem was that the label the service was using as selector was the same that the replicaset was using to reference its pods. Kubernetes will not allow you to modify or remove a template label if it is used to reference the pods, and will throw the following error:
`Patch failed: The ReplicaSet '' is invalid. spec.template.metadata.labels: Invalid value: ....... selector does not match template labels.`

TLDR; To solve this I just used another service selector label not used in the replicasets.
